### PR TITLE
Disallow radio usage on captivity and surrender

### PR DESCRIPTION
**Pull request by kwinteng**
**When merged this pull request will:**
- end the current radio transmission on captivity and surrender
- for both TFAR and ACRE disallow radio usage on captivity and surrender
- allow radio usage again on release/cancel surrender

### CODE REVIEW

File: addons/captives/functions/fnc_setHandcuffed.sqf [hunk 1]
```diff
@@ -41,6 +41,11 @@ if ((_unit getVariable [QGVAR(isHandcuffed), false]) isEqualTo _state) exitWith
 if (_state) then {
     _unit setVariable [QGVAR(isHandcuffed), true, true];
     [_unit, "setCaptive", QGVAR(Handcuffed), true] call EFUNC(common,statusEffect_set);
+    call FUNC(endRadioTransmission);
```
Comment (jonpas) [inline addons/captives/functions/fnc_setHandcuffed.sqf]: This should probably be `EFUNC(common,endRadioTransmission)` as it's locat.ed in `common` and you are doing this in `captives`. [This](http://ace3mod.com/wiki/development/coding-guidelines.html#macro-usage) should explain it.

File: addons/captives/functions/fnc_setHandcuffed.sqf [hunk 2]
```diff
@@ -41,6 +41,11 @@ if ((_unit getVariable [QGVAR(isHandcuffed), false]) isEqualTo _state) exitWith
...
+    //TFAR disallow radio usage
+    _unit setVariable ["tf_unable_to_use_radio", true];
+    //ACRE disallow radio usage
+    _unit setVariable ["acre_sys_core_isDisabled", true, true];
```
Comment (jonpas) [inline addons/captives/functions/fnc_setHandcuffed.sqf]: Why is TFAR's variable local and ACRE's broadcasted?

**Comment by jonpas** [comment]
Should probably do the same for `setHandcuffed`.

**Comment by kwinteng** [comment]
Fixed EFUNC()
added the same lines to surrender. @jonpas 
Still needs testing - will set up my test build asap

**Comment by esteldunedain** [comment]
I'm not sure if this handles properly the case in which a surrendered or handcuffed player dies and respawns.

**Comment by kwinteng** [comment]
I've just tested this manually, having some issues with setting up my dev build
1) the transmission cuts out when being captured
2) respawn recovers the radio usage
3) can't continuously transmit during surrendering so removing EFUNC(endtransmission) on surrendering

**Comment by Glowbal** [comment]
This change does not take into account other cases where the radio might have been disabled. This would also need to properly work with `ace_unconscious` We probably need to introduce some kind of helper method for doing this, similiar to our statusEffect solution.

**Comment by jonpas** [comment]
`setSurrendered`/`setHandcuffed` are called from `ace_onUnconscious` event.

**Comment by Glowbal** [comment]
@jonpas yes. It clears out the surrendered state for example; https://github.com/acemod/ACE3/blob/master/addons/captives/functions/fnc_handleOnUnconscious.sqf#L26

That would reset the radio usage variables for TFAR and ACRE.

**Comment by PabstMirror** [comment]
This might be a good case to use the status effect handler.
So both medical and captive don't overwrite eachother.

**Comment by commy2** [comment]
wip tag, cause it doesn't work as expected atm

**Comment by PabstMirror** [comment]
Closed via #8388

**Pull request closed without merging**